third_party/move/move-model/src/exp_rewriter.rs

```diff
@@ -3,17 +3,21 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{
-    ast::{Exp, ExpData, MemoryLabel, Operation, Pattern, Spec, TempIndex, Value},
+    ast::{Condition, Exp, ExpData, MemoryLabel, Operation, Pattern, Spec, TempIndex, Value},
     model::{GlobalEnv, ModuleId, NodeId, SpecVarId},
     symbol::Symbol,
     ty::Type,
 };
 use itertools::Itertools;
-use std::collections::BTreeSet;
+use log::trace;
+use move_binary_format::file_format::CodeOffset;
+use std::collections::{BTreeMap, BTreeSet};
 
 /// Rewriter for expressions, allowing to substitute locals by expressions as well as instantiate
-/// types.  Not safe for use to rewrite variables in expressions that may contain an `Assign`
-/// expression.
+/// types.
+///
+/// *** Note: will fail to rewrite variables when they appear in an `Assign` expression.***
```

Comment:
I had a check before, but @wrwg removed it for LambdaLifting, where it is OK to use since variables which are rewritten must not be modified, so can't appear in an `Assign` expression.  I've added a check now, but
it's pretty hard to test without a lot of plumbing/mocking.